apps/ios/Shared/Views/Chat/ChatItem/FramedItemView.swift

```diff
@@ -65,8 +66,13 @@ struct FramedItemView: View {
                                 }
                             }
                         }
-                } else if chatItem.meta.itemForwarded != nil {
-                    framedItemHeader(icon: "arrowshape.turn.up.forward", caption: Text("forwarded").italic())
+                } else if let itemForwarded = chatItem.meta.itemForwarded {
+                    if chat.chatInfo.chatType == .local,
+                       let fromChatName = itemForwarded.chatName {
+                        framedItemHeader(icon: "arrowshape.turn.up.forward", caption: Text("saved from \(fromChatName)").italic(), padBottom: true)
+                    } else {
+                        framedItemHeader(icon: "arrowshape.turn.up.forward", caption: Text("forwarded").italic(), padBottom: true)
```

Comment:
the condition can be just on text:

chat.chatInfo.chatType == .local ? "saved from \(fromChatName)" : "forwarded"

Reply:
and if it's local and there is no name it should be "saved" - probably better to get it to a function and avoid "if" here.

Reply:
first branch also requires checking fromChatName is not null

Reply:
refactored into function